public/app/plugins/panel/geomap/GeomapPanel.tsx

```diff
@@ -599,12 +604,19 @@ export class GeomapPanel extends Component<Props, State> {
         if (v.id === MapCenterID.Coordinates) {
           coord = [config.lon ?? 0, config.lat ?? 0];
         } else if (v.id === MapCenterID.Fit) {
-          const extent = getLayersExtent(layers);
+          const extent = getLayersExtent(layers, config.lastOnly ?? false);
           if (!isEmpty(extent)) {
+            const padding = config.padding ?? 5;
+            const res = view.getResolutionForExtent(extent, this.map?.getSize());
+            const maxZoom = config.zoom ?? config.maxZoom;
             view.fit(extent, {
-              padding: [30, 30, 30, 30],
-              maxZoom: config.zoom ?? config.maxZoom,
+              maxZoom: maxZoom,
             });
+            view.setResolution(res * (padding / 100 + 1));
```

Comment:
any insight on this calculation? (i.e. why padding / 100 + 1)

Reply:
Padding is a percentage > 0, so this is a multiplier to adjust the resolution accordingly. 50% for example would multiply resolution by 1.5 = 50 / 100 + 1.